client/cli/cli.go

```diff
@@ -139,10 +164,18 @@ func handleExit(dbp *proctl.DebuggedProcess, line *liner.State, status int) {
 		}
 	}
 
-	die(status, "Hope I was of service hunting your bug!")
+	t.die(status, "Hope I was of service hunting your bug!")
 }
 
-func die(status int, args ...interface{}) {
+type Terminator struct {
```

Comment:
I like this direction, but I think we should rename this type `Term`, and add more responsibility to it. I think that `promptForInput` should be a method off of this type, and we should add a field in the struct for `prompt`. That way, when we initialize it we could do so like: `Term{prompt: "(dlv) ", line: line}`.